src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -357,14 +367,29 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
    if (is.null(outputFile))
       outputFile <- .rs.withChangedExtension(inputFile, ext = ".nb.html")
 
-   # TODO: pass encoding from frontend
-   encoding <- "UTF-8"
+   # attempt to get encoding from source database (note: this will only
+   # succeed for files already open in the IDE, but since this operation
+   # is normally called when attempting to preview / create a notebook on
+   # save we generally expect the document to be available)
+   properties <- .rs.getSourceDocumentProperties(inputFile)
+   
+   # attempt to read and re-encode the file to UTF-8 if it's specified
+   # in the system encoding
+   if (!identical(properties$encoding, "UTF-8")) {
+      contents <- readLines(inputFile)
+      utf8 <- tryCatch(
+         iconv(contents, from = properties$encoding, to = "UTF-8"),
+         error = function(e) contents
```

Comment:
I don't know if the encodings accepted by `rmarkdown::render()` are the same as the encodings that we store as part of the `SourceDocument`'s properties, though.